audit-base/src/main/java/com/wultra/core/audit/base/database/DatabaseAuditWriter.java

```diff
@@ -159,6 +160,7 @@ public void flush() {
         }
 
         synchronized (FLUSH_LOCK) {
+            transactionTemplate.setPropagationBehavior(TransactionDefinition.PROPAGATION_REQUIRES_NEW);
```

Comment:
I have really tried to create a test, but the H2 engine does not respect the read-only option to simulate that.